src/wallet/wallet_tx_builder.cpp

```diff
@@ -177,14 +206,190 @@ SpendableInputs WalletTxBuilder::FindAllSpendableInputs(
     return wallet.FindSpendableInputs(selector, minDepth, std::nullopt);
 }
 
+static size_t NotOne(size_t n)
+{
+    return n == 1 ? 2 : n;
+}
+
+/// ISSUES:
+/// • z_shieldcoinbase currently checks the fee in advance of calling PrepareTransaction in order to
+///   determine the payment amount. But that fee possibly isn’t correct. We could perhaps have _no_
+///   payments in z_shieldcoinbase, but provide an explicit change address? (Don’t expose this via
+///   RPC, and give it a different name, e.g. `NetFundsRecipient` in the WalletTxBuilder interface)
+static CAmount
+CalcZIP317Fee(
+        const std::optional<SpendableInputs>& inputs,
+        const std::vector<ResolvedPayment>& payments,
+        const std::optional<std::optional<OutputPool>>& changeAddr)
```

Comment:
This is gone in the final commit as well. We now use the change address (instead of just the pool) because we need the size of the actual transparent address used in order to calculate the fee correctly.